lib/mshoplib/src/MShop/Plugin/Provider/Order/PropertyAdd.php

```diff
@@ -141,55 +141,35 @@ public function update( \Aimeos\MW\Observer\Publisher\Iface $order, string $acti
 	 * Adds the product properties as attribute items to the order product item
 	 *
 	 * @param \Aimeos\MShop\Order\Item\Base\Product\Iface $orderProduct Order product containing attributes
-	 * @param \Aimeos\Map $products List of items implementing \Aimeos\MShop\Product\Item\Iface with IDs as keys and properties
-	 * @param string[] $types List of property types to add
+	 * @param \Aimeos\Map $products list of items implementing \Aimeos\MShop\Product\Item\Iface with IDs as keys and properties
 	 * @return \Aimeos\MShop\Order\Item\Base\Product\Iface Modified order product item
 	 */
 	protected function addAttributes( \Aimeos\MShop\Order\Item\Base\Product\Iface $orderProduct,
-	\Aimeos\Map $products, array $types ) : \Aimeos\MShop\Order\Item\Base\Product\Iface
+	\Aimeos\Map $properties ) : \Aimeos\MShop\Order\Item\Base\Product\Iface
 	{
-		if( ( $product = $products->get( $orderProduct->getProductId() ) ) === null ) {
-			return $orderProduct;
-		}
-
-		if( $products->count() > 1 ) {
-			$variant = $products->col( null, 'product.code' )
-				->get( $orderProduct->getProductCode() );
-		}
-
-		foreach( $types as $type )
-		{
-			$list = $product->getProperties( $type );
+		$properties->each(function ( $attributes, $type ) use ( &$orderProduct ) {
```

Comment:
Where is `$type` comming from? `&` isn't necessary for objects

Reply:
`$type` is the key in the properties map, it's the type the user configures in the admin for the plugin.

An example of the properties passed into `addAttributes` is below.

```php
Array
(
    [package-width] => Aimeos\Map Object
        (
            [list:protected] => Array
                (
                    [7] => 15.0
                )

            [sep:protected] => /
        )

)
```